publish_data.go

```diff
@@ -38,21 +39,54 @@ func (dp *DataPublisher) RemoveLJH22() {
 	dp.LJH22.Close()
 	dp.LJH22 = nil
 }
+func (dp *DataPublisher) HasPubFeederChan() bool {
+	return dp.PubFeederChan != nil
+}
+
+func (dp *DataPublisher) SetPubFeederChan() {
+	dp.PubFeederChan = PubFeederChan
+}
+func (dp *DataPublisher) RemovePubFeederChan() {
+	dp.PubFeederChan = nil
+}
+
+var PubFeederChan chan []*DataRecord
```

Comment:
Why do you have this as a package-global variable, and then you sometimes use it within each DataPublisher and sometimes don't? Why not just have a DataPublisher attribute (or whatever go calls them) that's a bool to indicate whether to use it? 

Probably because multiple DataPublishers need to share this channel, but this somehow feels like the wrong way. I can't really put my finger on it.